specs/albums.spec.js

```diff
@@ -46,79 +45,80 @@ describe("ALBUMS", () => {
         expect(album).to.be.defined;
         expect(album).to.deep.equal(expectedAlbum);
       });
-      return chakram.wait();
     });
 
-    it("should not return an album with non-existing ID", () => {
-      const response = chakram.get(api.url("albums/1234567"));
+    it("should not return an album with non-existing ID", async () => {
+      const response = await chakram.get(api.url("albums/1234567"));
       expect(response).to.have.status(404);
-      return chakram.wait();
     });
   });
 
   describe("UPDATE", () => {
-    it("should update az existing album", () => {
+    it("should update an existing album", async () => {
+      const albumId = Math.floor(Math.random() * data.albums.length);
       const albumUpdate = {
         userId: 1,
-        id: 1,
+        id: albumId,
         title: "updated title",
       };
-      const albumId = 1;
-      const response = chakram.put(api.url(`albums/${albumId}`), albumUpdate);
+      const response = await chakram.put(api.url(`albums/${albumId}`), albumUpdate);
       expect(response).to.have.status(200);
-      const updatedAlbum = chakram.get(api.url(`albums/${albumId}`));
+      const updatedAlbum = await chakram.get(api.url(`albums/${albumId}`));
       expect(updatedAlbum).have.json("data", album => {
         expect(album).to.be.defined;
         expect(album.title).to.equal(albumUpdate.title);
         expect(album.id).to.equal(albumUpdate.id);
         expect(album.userId).to.equal(albumUpdate.userId);
       });
-      return chakram.wait();
     });
 
-    it("should not update an album which does not exist", () => {
+    it("should not update an album which does not exist", async () => {
       const albumUpdate = {
         userId: 1,
         id: 1,
         title: "updated title",
       };
-      const response = chakram.put(api.url("albums/180"), albumUpdate);
+      const response = await chakram.put(api.url("albums/180"), albumUpdate);
       expect(response).to.have.status(404);
-      return chakram.wait();
     });
   });
 
   describe("DELETE", () => {
-    it("should delete an existing album", () => {
-      const response = chakram.delete(api.url("albums/3"));
-      expect(response).to.have.status(200);
-      const notExistingAlbum = chakram.get(api.url("albums/3"));
+    it("should delete an existing album", async () => {
+      const originalLength = data.albums.length;
+      const responseDel = await chakram.delete(api.url("albums/3"));
+      expect(responseDel).to.have.status(200);
+      const notExistingAlbum = await chakram.get(api.url("albums/3"));
       expect(notExistingAlbum).to.have.status(404);
-      return chakram.wait();
+
+      const responseGet = await chakram.get(api.url("albums"));
+      expect(responseGet).to.have.status(200);
+      expect(responseGet).to.have.json("data", albums => {
+        expect(albums).to.be.instanceOf(Array);
+        expect(albums.length).not.to.equal(originalLength);
```

Comment:
It would be a stricter expectation. You can check that a number is less than the given number.

Reply:
made it stricter